bruig/flutterui/bruig/lib/components/usersearch/user_search_panel.dart

```diff
@@ -0,0 +1,319 @@
+        oldWidget.hintText != widget.hintText) {
+      controller.text = "";
+    }
+  }
+
+  void handleKeyPress(KeyEvent event) {
+    bool modPressed = HardwareKeyboard.instance.isShiftPressed ||
+        HardwareKeyboard.instance.isControlPressed;
+    widget.onChanged(controller.text);
+    if (event.logicalKey.keyLabel == "Enter" && !modPressed) {
+      controller.value = const TextEditingValue(
+          text: "", selection: TextSelection.collapsed(offset: 0));
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    bool isScreenSmall = checkIsScreenSmall(context);
+    return KeyboardListener(
+        focusNode: node,
+        onKeyEvent: handleKeyPress,
+        child: Container(
+            margin: const EdgeInsets.only(bottom: 5),
+            child: Row(children: [
+              Expanded(
+                child: TextField(
+                  autofocus: isScreenSmall ? false : true,
+                  focusNode: widget.inputFocusNode.inputFocusNode,
+                  controller: controller,
+                  minLines: 1,
+                  maxLines: null,
+                  keyboardType: TextInputType.multiline,
+                  decoration: InputDecoration(
+                    isDense: true,
+                    hintText: widget.hintText ??
+                        'Search name of user ${widget.onlyUsers ? "" : "or group chat"}',
+                    border: const OutlineInputBorder(
+                        borderRadius: BorderRadius.all(Radius.circular(30)),
+                        borderSide: BorderSide(width: 1)),
+                  ),
+                ),
+              )
+            ])));
+  }
+}
+
+enum UserSearchPanelTargets {
+  usersAndGCs,
+  users,
+  gcs,
+}
+
+class UserSearchPanel extends StatefulWidget {
+  final ClientModel client;
+  final UserSearchPanelTargets targets;
+  final CustomInputFocusNode? inputFocusNode;
+  final VoidCallback? onCancel;
+  final VoidCallback? onConfirm;
+  final ValueChanged<ChatModel>? onChatTapped;
+  final String confirmLabel;
+  final UserSelectionModel? userSelModel;
+  final bool showButtonsRow;
+  final String? searchInputHintText;
+  final ValueChanged<String>? onSearchInputChanged;
+  final List<ChatModel>? sourceChats;
+  const UserSearchPanel(
+    this.client, {
+    super.key,
+    this.targets = UserSearchPanelTargets.users,
+    this.confirmLabel = "Confirm",
+    this.inputFocusNode,
+    this.userSelModel,
+    this.showButtonsRow = true,
+    this.searchInputHintText,
+    this.onCancel,
+    this.onConfirm,
+    this.onChatTapped,
+    this.onSearchInputChanged,
+    this.sourceChats,
+  });
+
+  @override
+  State<UserSearchPanel> createState() => _UserSearchPanelState();
+}
+
+class _UserSearchPanelState extends State<UserSearchPanel> {
+  ClientModel get client => widget.client;
+  List<ChatModel> chats = [];
+  late CustomInputFocusNode inputFocusNode;
+  String filterSearchString = "";
+  List<ChatModel> filteredSearch = [];
```

Comment:
Noticed that current members of a GC were included in the list to Invite New Members.  So maybe add a passed initial state here to filter those names out of the list?